### PR TITLE
Update misleading pegout log messages and variables

### PR DESCRIPTION
**Pull request by nathanieliov**
- Update misleading pegout logs with meaningful log messages according to the context.
- Also update misleading pegout variable names with meaningful names according to the context.

### CODE REVIEW

**Comment by nathanieliov** [comment]
pipeline:run

**Comment by nathanieliov** [comment]
https://capitan.wefrkpjl.name/job/Pipeline/11258/

**Review by jeremy-then — DISMISSED** [review]
LGTM.

File: src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java [hunk 1]
```diff
@@ -216,22 +216,22 @@ public void onBlock(org.ethereum.core.Block block, List<TransactionReceipt> rece
...
             Stream<LogInfo> bridgeLogs = transactionLogs.filter(info -> Arrays.equals(info.getAddress(), PrecompiledContracts.BRIDGE_ADDR.getBytes()));
 
             boolean solidityFormatIsActive = activationConfig.isActive(ConsensusRule.RSKIP146, block.getNumber());
-            Stream<LogInfo> releaseBtcLogs = bridgeLogs.filter(info -> solidityFormatIsActive ?
+            Stream<LogInfo> pegoutLogs = bridgeLogs.filter(info -> solidityFormatIsActive ?
                     SINGLE_RELEASE_BTC_TOPIC_SOLIDITY.equals(info.getTopics().get(0)) :
                     SINGLE_RELEASE_BTC_TOPIC_RLP.equals(info.getTopics()));
 
-            Stream<BtcTransaction> btcTransactionsToRelease = releaseBtcLogs.map(info -> solidityFormatIsActive ?
+            Stream<BtcTransaction> pegoutBtcTxs = pegoutLogs.map(info -> solidityFormatIsActive ?
```
Comment (julia-zack) [inline src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java:230]: I think there's no need to add `Btc` to the variable name, i'd just name it as `pegoutTxs`

File: src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java [hunk 2]
```diff
@@ -246,71 +246,78 @@ private BtcTransaction convertToBtcTxFromSolidityData(byte[] dataFromBtcReleaseT
...
-            // Get release information and store it in a new list
-            List<ReleaseCreationInformation> releasesReadyToSign = new ArrayList<>();
-            for (Map.Entry<Keccak256, BtcTransaction> release : releases) {
-                BtcTransaction releaseTx = release.getValue();
-                tryGetReleaseInformation(version, release.getKey(), releaseTx)
-                    .ifPresent(releasesReadyToSign::add);
+            // Get pegout information and store it in a new list
+            List<ReleaseCreationInformation> pegoutsReadyToSign = new ArrayList<>();
+            for (Map.Entry<Keccak256, BtcTransaction> pegout : pegouts) {
+                /*
```
Comment (julia-zack) [inline src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java:256]: Having the comment, I think the `Creation` part is no needed (and a bit confusing) in the variable. i'd simply call it `pegoutRskTxHash`. 

(btw super useful comment, well done!)

File: src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java [hunk 3]
```diff
@@ -246,71 +246,78 @@ private BtcTransaction convertToBtcTxFromSolidityData(byte[] dataFromBtcReleaseT
...
+
+                tryGetReleaseInformation(version, pegoutCreationRskTxHash, pegoutBtcTx)
+                    .ifPresent(pegoutsReadyToSign::add);
             }
-            logger.debug("[processReleases] Going to sign {} releases", releasesReadyToSign.size());
+            logger.debug("[processReleases] Going to sign {} pegouts", pegoutsReadyToSign.size());
             // TODO: Sorting and then looping again is not efficient but we are making a compromise on performance here as we don't have that many release txs
             // Sort descending
-            releasesReadyToSign.sort((a, b) -> (int) (b.getBlock().getNumber() - a.getBlock().getNumber()));
+            pegoutsReadyToSign.sort((a, b) -> (int) (b.getPegoutCreationRskBlock().getNumber() - a.getPegoutCreationRskBlock().getNumber()));
```
Comment (julia-zack) [inline src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java:269]: same here, let's call it `getPegoutRskBlock`

File: src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java [hunk 4]
```diff
@@ -246,71 +246,78 @@ private BtcTransaction convertToBtcTxFromSolidityData(byte[] dataFromBtcReleaseT
...
-                storageAccessor.getRskTxHash(releaseTx.getHash()) :
-                rskTxHash;
+            logger.trace("[tryGetReleaseInformation] Is pegout btc in storage? {}", storageAccessor.hasBtcTxHash(pegoutBtcTx.getHash()));
+            // Try to get the pegoutCreationRskTxHash from the map in memory
+            Keccak256 pegoutCreationRskTxHashToUse = storageAccessor.hasBtcTxHash(pegoutBtcTx.getHash()) ?
+                storageAccessor.getRskTxHash(pegoutBtcTx.getHash()) :
+                pegoutCreationRskTxHash;
 
-            logger.debug("[tryGetReleaseInformation] Going to lookup tx to sign {}", actualRskTxHash);
+            logger.debug("[tryGetReleaseInformation] Going to lookup pegoutCreationRskTxHash {} to sign", pegoutCreationRskTxHash);
```
Comment (julia-zack) [inline src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java:306]: shouldn't it be the `pegoutRskTxHashToUse` ?

File: src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientStorageAccessor.java [hunk 5]
```diff
@@ -63,7 +63,7 @@ public BtcReleaseClientStorageAccessor(
                 throw new InvalidStorageFileException(message, e);
             }
         }
-        if (!readResult.getSuccess()) {
+        if (Boolean.FALSE.equals(readResult.getSuccess())) {
```
Comment (julia-zack) [inline src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientStorageAccessor.java:66]: why this change? i think we are not doing that anywhere else

Comment (nathanieliov) [reply src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientStorageAccessor.java:66]: To solve a Sonar complaint.

image

File: src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java [hunk 6]
```diff
@@ -357,31 +364,32 @@ protected void validateTxCanBeSigned(BtcTransaction btcTx) throws FederatorAlrea
...
 
-    protected void signRelease(int signerVersion, ReleaseCreationInformation releaseCreationInformation) {
+    protected void signRelease(int signerVersion, ReleaseCreationInformation pegoutCreationInformation) {
+        final String topic = "btcrelease";
         try {
-            logger.debug("[signRelease] HSM signer version {}", signerVersion);
-            logger.debug("[signRelease] Going to sign tx {}", releaseCreationInformation.getInformingRskTxHash());
-            logger.trace("[signRelease] Enforce signer requirements");
-            releaseRequirementsEnforcer.enforce(signerVersion, releaseCreationInformation);
+            logger.debug("[signConfirmedPegout] HSM signer version {}", signerVersion);
```
Comment (marcos-iov) [inline src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java:382]: Did the method name change?

File: src/main/java/co/rsk/federate/bitcoin/BitcoinWrapperImpl.java [hunk 7]
```diff
@@ -349,7 +349,7 @@ protected void coinsReceivedOrSent(Transaction tx) {
                     listener.onTransaction(tx);
                 }
                 if (PegUtilsLegacy.isPegOutTx(btcTx, Collections.singletonList(watchedFederation), federatorSupport.getConfigForBestBlock())) {
-                    LOGGER.debug("[coinsReceivedOrSent] [btctx:{}] is a release", tx.getWTxId());
+                    LOGGER.debug("[coinsReceivedOrSent] [btctx:{}] is a pegout", tx.getWTxId());
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply src/main/java/co/rsk/federate/bitcoin/BitcoinWrapperImpl.java:352]: Sounds good to me. Just mention that for non segwit txs it is the same.

File: src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientStorageAccessor.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (nathanieliov) [reply src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClientStorageAccessor.java:66]: To solve a Sonar complaint. As mentioned above.

File: src/main/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformation.java [hunk 8]
```diff
@@ -6,74 +6,73 @@
 import org.ethereum.core.TransactionReceipt;
 
 public class ReleaseCreationInformation {
-    private final Block block;
+    private final Block pegoutCreationRskBlock;
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply src/main/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformation.java:9]: That's a valid argument, we just need to keep in mind to name it the same way in all the places. The reason why I did like that was to be consistent with the `PegoutsWaitingForConfirmations` entry class https://github.com/rsksmart/rskj/blob/5a83e62d0c17805679eccae6e958e1f8973b7d57/rskj-core/src/main/java/co/rsk/peg/PegoutsWaitingForConfirmations.java#L48.

File: src/main/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformation.java [hunk 9]
```diff
@@ -6,74 +6,73 @@
...
     private final TransactionReceipt transactionReceipt;
-    private final Keccak256 releaseRskTxHash;
-    private final Keccak256 informingRskTxHash;
-    private final BtcTransaction btcTransaction;
-
+    private final Keccak256 pegoutCreationRskTxHash;
+    private final BtcTransaction pegoutBtcTx;
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply src/main/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformation.java:12]: So, from now on, we are always going to refer to pegout btc transaction objects as `pegoutTx`, right?

File: src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java (hunk 6 above)
Comment (nathanieliov) [reply src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java]: It didn't. It came from the previous PR. Just fixed it.

File: src/main/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformationGetter.java [hunk 10]
```diff
@@ -83,33 +83,33 @@ protected ReleaseCreationInformation getBaseReleaseCreationInformation(
...
             ReleaseCreationInformation baseReleaseCreationInformation =
-                getBaseReleaseCreationInformation(rskTxHash, btcTransaction, informingRskTxHash);
-            Block block = baseReleaseCreationInformation.getBlock();
+                getBaseReleaseCreationInformation(pegoutCreationRskTxHash, pegoutBtcTx, pegoutConfirmationRskTxHash);
+            Block block = baseReleaseCreationInformation.getPegoutCreationRskBlock();
             TransactionReceipt transactionReceipt = baseReleaseCreationInformation.getTransactionReceipt();
 
             // Get transaction from the block, searching by tx hash, and set it in the tx receipt
-            logger.trace("[getTxInfoToSign] Searching for transaction {} in block {} ({})", rskTxHash, block.getHash(), block.getNumber());
+            logger.trace("[getTxInfoToSign] Searching for pegoutCreationRskTxHash {} in block {} ({})", pegoutCreationRskTxHash, block.getHash(), block.getNumber());
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply src/main/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformationGetter.java]: Fixed.

**Comment by marcos-iov** [comment]
pipeline:run

**Comment by nathanieliov** [comment]
This PR was replaced by PR #251
